### ISSUE TITLE
Error occurred during the conversion of other versions (s, m, l, x) to the imx format, which is not Yolov8n.

### ISSUE THREAD

**Issue by sevdekutuk** [ultralytics/ultralytics#18440]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I am getting an error while converting other versions (s, m, l, x) to imx format except Yolov8n. Nano weight is converted and used, but I am getting the following error for other weights. What should I do?
<img width="793" alt="Ekran Resmi 2024-12-29 14 55 08" src="https://github.com/user-attachments/assets/86d80307-1447-45c5-ac59-958db02f3633" />
<img width="631" alt="Ekran Resmi 2024-12-29 14 55 35" src="https://github.com/user-attachments/assets/7f6b348d-f490-48c1-a0ab-d575d78abd43" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sevdekutuk, thank you for reporting this issue and your interest in Ultralytics 🚀! We appreciate your detailed description and screenshots.

To assist us in diagnosing the problem, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) that includes the code snippet, command, or any configuration you are using to convert the models to the `imx` format. This will help us reproduce and investigate the error effectively 🐛.

Here are some steps to make sure you’re using the latest setup:
1. **Upgrade to the latest `ultralytics` package**:
   Ensure you are using the latest release, as your issue may already be resolved. To upgrade:
   ```bash
   pip install -U ultralytics
   ```
2. **Environment and Dependencies**:
   Use an up-to-date Python environment with all dependencies satisfied. Ensure you have [**Python>=3.8**](https://www.python.org/) installed along with the latest [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Resources
For further guidance, a review of our documentation might be helpful. You can find detailed usage examples here:
- [Documentation](https://docs.ultralytics.com)
- [CLI Usage](https://docs.ultralytics.com/usage/cli/)
- [Model Formats](https://docs.ultralytics.com/modes/export/) (specific to export and various formats)

## Verified Environments
We recommend using any of the following verified environments for consistent results. They come equipped with the necessary dependencies pre-installed:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Other** deployments: Check out guides for [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/), [GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/), and [Docker Image](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## Community Support
Feel free to engage with the Ultralytics community for additional help:
- Join real-time discussions on [Discord](https://discord.com/invite/ultralytics) 🎧
- Post questions or detailed issues on [Discourse](https://community.ultralytics.com) 
- Explore and share insights on our [Subreddit](https://reddit.com/r/Ultralytics)

Thank you again for bringing this to our attention! This is an automated response 🤖, but rest assured that an Ultralytics engineer will follow up with further assistance soon.

**Comment by Y-T-G** [comment]
IMX only supports YOLOv8n

https://github.com/sony/model_optimization/issues/1269#issuecomment-2483250853

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in [PR #18460](https://github.com/ultralytics/ultralytics/pull/18460) 🎉. 

### Key Changes from the PR:
- **Improved Clarity for IMX Export**: Updated documentation and error messages to specify that IMX export is only supported for YOLOv8n detection models.  
- **Compatibility Assurance**: Limitations are now explicit, helping prevent confusion and ensuring smoother workflows for users deploying on hardware like the Raspberry Pi AI Camera.  

Huge thanks to the PR author [@Y-T-G](https://github.com/Y-T-G) and contributor [@ambitious-octopus](https://github.com/ambitious-octopus) for their efforts in refining this functionality! 🙌

### Testing the Fix:
You can test the latest changes by running:
```bash
pip install git+https://github.com/ultralytics/ultralytics.git@main
```
Alternatively, feel free to wait for the next official PyPI release.

We’d greatly appreciate it if you could confirm whether this resolves your issue. If the problem persists or you have additional feedback, don’t hesitate to share—it’s invaluable for improving the project.

Thanks so much for reporting this and helping make the project better! 🙏
